apps/storybook/package.json

```diff
@@ -37,7 +37,9 @@
     "build": "dotenv -v NODE_OPTIONS=\"--max-old-space-size=4096\" sb build",
     "dev": "sb dev -p 6006 --no-open --quiet",
     "lint": "eslint \"**/*.{ts,tsx}\" --max-warnings=0",
-    "test": "node scripts/run-tests.js",
+    "test": "yarn test:visual && yarn test:types",
+    "test:visual": "node scripts/run-tests.js",
+    "test:types": "tsc -p tsconfig.test.json --noEmit",
```

Comment:
this is not a test. it should probably be part of `build`?

Reply:
I intentionally added the [`--noEmit`](https://www.typescriptlang.org/tsconfig#noEmit) flag so that it does not output anything. Thus, this serves as just a type test.

Reply:
yes, that should go in `build`. if the build passes, that means types are good. you can see an example in `playgrounds/vite/package.json`

Reply:
Agree that types should be checked in build.

Reply:
This approach will work well for files that are part of the build output. However, this does not check for type errors in non-build files, like the unit test files.

Apart from the unit test files, I also have one file called [`react-table-types.tsx`](https://github.com/iTwin/iTwinUI/blob/f7140d13e40d511ac8ff77e6150e93f270effbab/packages/itwinui-react/src/core/Table/types/react-table-types.test.tsx) that is not a part of the build. This file is to test the types in a few unique cases.

Reply:
i don't understand. this thread is about running `tsc` in storybook. all i'm saying is this should be done as part of `build`, because `test` is for running visual tests only

```diff
-    "build": "dotenv -v NODE_OPTIONS=\"--max-old-space-size=4096\" sb build",
+    "build": "tsc && dotenv -v NODE_OPTIONS=\"--max-old-space-size=4096\" sb build",=
     ...
     "test": "node scripts/run-tests.js",
```

Reply:
Sure, then for storybook, I combined the types tests into the build script (https://github.com/iTwin/iTwinUI/pull/1514/commits/fd13f3a3118e7055507b016fa99aad0624d9da51)

I temporarily only test for `Table.stories.tsx` since other stories have many errors. I added a `TODO` to explain that. In a future PR, we can fix those errors and remove that `TODO`.

But for `packages/itwinui-react`, I still kept the `test` script to test `react-table-types.tsx`. In the future, we can extend it to test *all* files in the package, and *not* just files in the build pipeline.

Reply:
i think you forgot to bring back the old `"test"` script (it's now pointing to a non-existent `yarn test:visual` script)

Reply:
https://github.com/iTwin/iTwinUI/pull/1514/commits/230b73477716175e772c7206bf0bf4170d4d1536: Corrected